ADLStream/evaluation/base_evaluator.py

```diff
@@ -127,15 +137,19 @@ def evaluate(self):
 
     def write_results(self, new_results, instances):
         if self.results_file is not None:
-            with open(self.results_file, "a") as f:
-                for i, value in enumerate(new_results):
-                    f.write(
-                        "{},{},{}\n".format(
-                            str(datetime.now()),
-                            instances[i],
-                            value,
-                        )
+            for i, value in enumerate(new_results):
+                self.results_file.write(
+                    "{},{},{}\n".format(
+                        str(datetime.now()),
+                        instances[i],
+                        value,
                     )
+                )
+
+    def write_predictions(self):
+        if self.predictions_file is not None:
+            for _, prediction in enumerate(self.o_eval):
```

Comment:
Instead of saving `o_eval` we should save only the new predictions (variable `o` from line 164)
```
    def write_predictions(self, preds):
        if self.predictions_file is not None:
            for _, prediction in enumerate(preds):
                ...
```
```
    def update_predictions(self, context):
        x, y, o = context.get_predictions()
        ...
        self.write_predictions(o)
```